BatchApex/BatchLeads.cls

```diff
@@ -0,0 +1,75 @@
+/**
+ * Removal of duplicate Leads 
+ * During the Campaigning, It might happen that representative creates duplicate leads in an org. 
+ * So admin want to build a process which will run every 3 hours/day &  
+ * remove the duplicate leads from the org. 
+ * The criteria to find the duplicate records should be configurable. 
+ */
+
+global with sharing class BatchLeads implements Database.Batchable<sObject>, Database.Stateful {
```

Comment:
from Next tym please add properr class names. What does Batch Leads mean? this batch class is to remove duplicate leads, if we read this class name we can't understand wht you are actually doing in this class. Name should be something like RemoveDuplicateLeadsBatch